torch/utils/_mode_utils.py

```diff
@@ -71,7 +71,7 @@ def _enable_mode(mode, mode_info: _ModeInfo, *, replace=None, ignore_preexisting
         raise ValueError(f'expected to get {mode_info.mode_class_name()}, Tensor-like class, '
                          f'or None as an argument got {type(mode)} instead')
     old = mode_info.get_mode()
-    if old is mode:
+    if type(old) is type(mode):
```

Comment:
This may no longer support the old version of python mode

Reply:
Just checking, this still needs to be fixed because we can have multiple instances of the same class, right? (When this does get fix we should add a test for it)